lib/stdio/generator.js

```diff
@@ -36,7 +36,7 @@ The same applies to the first output's generator's `writableObjectMode`.
 */
 const getOutputObjectModes = (objectMode, index, newGenerators) => {
 	const writableObjectMode = index !== 0 && newGenerators[index - 1].value.readableObjectMode;
-	const readableObjectMode = objectMode;
+	const readableObjectMode = objectMode === 'previous' ? writableObjectMode : objectMode;
```

Comment:
For output streams, `writableObjectMode` is always the same as the previous generator's `readableObjectMode` (if any).